### PR TITLE
NEW sqlalchemy tables for scores and parameters 

### PR DESCRIPTION
**Pull request by gwarmstrong**
This PR adds a database module to q2-mlab, to make for convenient access to the bench-marking results.

This first pass, uses [`sqlalchemy`](https://www.sqlalchemy.org/) to create the table schemas and interact with the database. I have also exposed some convenience methods for adding rows to the database via a cli.

### CODE REVIEW

**Comment by gwarmstrong** [comment]
Looks like there was some breaking change made in the latest new major version of `lightgbm`, so I pinned the version for now.

File: q2_mlab/db/maint.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+
+def add(engine: Engine, results: pd.DataFrame, parameters: dict,
+        dataset: str, target: str, level: str, algorithm: str,
+        ) -> None:
+    Session = sessionmaker(bind=engine)
+    session = Session()
+
+    params = Parameters(**parameters)
+    session.add(params)
```
Comment (gwarmstrong) [inline q2_mlab/db/maint.py]: Check for these exact parameters existing in database

File: q2_mlab/db/maint.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    session.flush()
+
+    for entry in results.iterrows():
+        score = RegressionScore(datetime=datetime.now(),
+                                parameters_id=params.id,
+                                dataset=dataset,
+                                target=target,
+                                level=level,
+                                algorithm=algorithm,
```
Comment (gwarmstrong) [inline q2_mlab/db/maint.py:69]: Check if the algorithm is valid?

File: q2_mlab/db/maint.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        ) -> None:
+    Session = sessionmaker(bind=engine)
+    session = Session()
+
+    params = Parameters(**parameters)
+    session.add(params)
+    session.flush()
+
+    for entry in results.iterrows():
+        score = RegressionScore(datetime=datetime.now(),
```
Comment (gwarmstrong) [inline q2_mlab/db/maint.py]: Push `datetime` up to before for loop

File: q2_mlab/db/maint.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                 level: str,
+                 algorithm: str,
+                 db_file: Optional[str] = None,
+                 echo: bool = True,
+                 engine_creator: Callable = create_engine,
+                 ):
+
+    engine = engine_creator(db_file, echo=echo)
+
+    results = Artifact.load(artifact_path).view(pd.DataFrame)
```
Comment (gwarmstrong) [inline q2_mlab/db/maint.py]: Get uuid from the artifact?

```suggestion
    artifact = Artifact.load(artifact_path)
    artifact_uuid = artifact.uuid
    results = artifact.view(pd.DataFrame)
```

File: q2_mlab/db/maint.py (hunk 3 above)
Comment (gwarmstrong) [inline q2_mlab/db/maint.py]: Store `artifact_uuid`

File: q2_mlab/db/schema.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+Base = declarative_base()
+
+
+class Score:
+    id = Column(Integer, primary_key=True)
+    algorithm = Column(String, nullable=False)
+    dataset = Column(String, nullable=False)
+    level = Column(String, nullable=False)  # 16S, metagenomics, etc.
+    target = Column(String, nullable=False)
+    datetime = Column(DateTime)
```
Comment (gwarmstrong) [inline q2_mlab/db/schema.py:24]: Add `artifact_uuid`

File: q2_mlab/db/tests/test_mapping.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+
+class ParameterRemapTestCase(unittest.TestCase):
+    def test_trivial_remapper(self):
+        parameter = 'param'
+        value = 24
+        obs = trivial_remapper(parameter, value)
+        self.assertDictEqual({'param': 24}, obs)
+
+    def test_string_or_number_remapper(self):
```
Comment (patrickimran) [inline q2_mlab/db/tests/test_mapping.py:18]: Let's add a test for None parameters, or maybe cast those to strings.

`string_or_number_remapper` is giving a `TypeError: 'NoneType' object is not iterable` when the parameter is None, as in this example:
`{'bootstrap': False, 'criterion': 'mae', 'max_depth': 7, 'max_features': None, 'max_samples': 0.75, 'min_samples_leaf': 0.41000000000000003, 'min_samples_split': 2, 'n_estimators': 1000, 'n_jobs': -1, 'random_state': 2020}`
